config/checkstyle-examples-checks.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+  <property name="haltOnException" value="false"/>
+
+  <property name="cacheFile" value="target/cache_xdocs_examples"/>
+
+  <module name="SuppressionFilter">
+    <property name="file" value="${checkstyle.examples-suppressions.file}"/>
+  </module>
+
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="false"/>
+  </module>
+
+  <module name="FileLength">
+    <property name="max" value="45"/>
+  </module>
+
+  <module name="RegexpOnFilename">
+    <property name="id" value="exampleFileName"/>
+    <property name="fileNamePattern" value="^Example\d+"/>
+    <property name="ignoreFileNameExtensions" value="true"/>
+    <property name="match" value="false"/>
+    <message key="regexp.filename.mismatch" value="All example files should start with Example\d+"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="noTrailingWhitespace"/>
+    <property name="format" value="\s+$"/>
+    <property name="minimum" value="0"/>
+    <property name="maximum" value="0"/>
+    <property name="message" value="Trailing whitespace is not allowed"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="noDefaultProperties"/>
+    <property name="format" value=".+\s=\s\(default\).*"/>
+    <property name="message" value="(default) properties are not allowed"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="xdocStartComment"/>
+    <property name="format" value="// xdoc section \-\- start"/>
+    <property name="message" value="Missing comment '// xdoc section -- start'"/>
+    <property name="minimum" value="1"/>
+    <property name="maximum" value="1"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="xdocEndComment"/>
+    <property name="format" value="// xdoc section \-\- end"/>
+    <property name="message" value="Missing comment '// xdoc section -- end'"/>
+    <property name="minimum" value="1"/>
+    <property name="maximum" value="1"/>
+  </module>
+
+  <module name="LineLength">
+    <property name="max" value="85"/>
+  </module>
+
+  <module name="TreeWalker">
+    <module name="Indentation">
```

Comment:
Add `CommentsIndentation`.

Reply:
Better to add it later on, please create issue to do this in separate PR.
In other case we will never finish first step - just some validation activation.

Reply:
Added to https://github.com/checkstyle/checkstyle/issues/13244#issuecomment-1600605964. We are going to add more stuff the moment this PR is merged.